### ISSUE TITLE
[a11y]: TableToolbar: inappropriate region role

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/carbon#20623]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.89.0

### React version

18

### Automated testing tool and ruleset

n/a

### Assistive technology

no

### Description

Carbon currently renders the `TableToolbar` as a `<section>` with an `aria-label`, either a user-defined value or the default value of "data table toolbar".

This means that the `TableToolbar` is rendered with an implicit `region` role.  I think it's inappropriate for such a small thing to be rendered as a region:

According to [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/region_role):

> The region role is an [ARIA landmark](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles#3._landmark_roles) role. The region role should be reserved for sections of content sufficiently important that users will likely want to navigate to the section easily and to have it listed in a summary of the page.

And similarly, [W3](https://www.w3.org/TR/WCAG20-TECHS/ARIA20.html) says that:

> The role of region should be used prudently, because if overused they can make the page overly verbose for screen reader users.

It would be nice to use role="toolbar" but the searchbox in the middle of the toolbar would make left-right arrow key navigation difficult.

I think any of these renderings would be more appropriate:

```html
<div role="group" aria-label="data table toolbar">
<section role="group" aria-label="data table toolbar">
<fieldset> <legend class="cds--visually-hidden">data table toolbar</legend> … </fieldset>
```

### WCAG 2.1 Violation

https://www.w3.org/TR/WCAG21/#name-role-value

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-datatable-toolbar--default

### Steps to reproduce

Just look at the generated HTML.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by nehaaprasad** [comment]
@wkeese 
I've fixed it by adding role='group' to prevent the inappropriate region landmark behavior

**Comment by wkeese** [comment]
Great, thanks!
